src/lib/linear_model/logistic_regression.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  mul,
+  randomUniform,
+  Rank,
+  round,
+  sigmoid,
+  Tensor,
+  Tensor1D,
+  tensor1d,
+  tensor2d
+} from '@tensorflow/tfjs';
+import { Type1DMatrix, Type2DMatrix } from '../types';
+import { ensure2DMatrix, inferShape } from '../utils/tensors';
+import { checkNumFeatures, validateFitInputs } from '../utils/validation';
+
+export class LogisticRegression {
+  private weights: Tensor1D;
+  private learningRate: number;
+
+  constructor(learningRate = 0.1) {
```

Comment:
we generally use object-based param for the constructor.

Reply:
Also, we should at least expose an option `solver` so we can extend it to support newton solver later